### ISSUE TITLE
Any thoughts on a command line compiler templates > compiles javascript?

### ISSUE THREAD

**Issue by voidfiles** [akdubya/dustjs#13]
I am looking at using DustJS as at templating library. I would really like to compile templates down to javascript from the command line, and then include them in by static asset build system.

Does anyone have an thoughts on how this could work?

**Comment by xonecas** [comment]
Yeah, the way I got that to work, is to use a pre start script that reads all of the templates and then compiles and writes them to a static .js file. To use from the command line, just use node.js cli.

**Comment by voidfiles** [comment]
I didn't see a compile script is there one already?

**Comment by xonecas** [comment]
You'll have to write your own.

**Comment by voidfiles** [comment]
Let us know if anyone is interested but we ended up creating a system for using compiled templates in production, but compiling them on demand in dev.

**Comment by copenhas** [comment]
Doesn't look like there has been any activity on this. I started playing around with dust.js a month ago or so and was using compiled templates both in CouchDB and served up as static javascript. For this I wrote a script to do it, but recently started trying to take my script to the "next level". It's on github, so anyone who may find it useful by all means. I have some plans for it to help me learn a bit more about the node landscape (vows.js specs, publish to npm, real CLI interface, better logging, the usual).

https://github.com/copenhas/broom

**Comment by deoxxa** [comment]
I did the same thing as @copenhas with a slightly different approach, mine is just a simple utility to (recursively) compile a directory of templates. There's no configuration, and you can include it as a library to wrap your own interface around it (an express middleware has been one suggestion).

https://github.com/deoxxa/dusty
